equipment_handler/source/ina226.c

```diff
@@ -49,40 +48,7 @@ int INA226_RegisterSet(i2cBASE_t *i2c,
   data[0] = val >> 8;
   data[1] = val & 0xFF;
 
-
-//  //Check if the busy is currently busy
-//  if((i2c->STR & (uint32)I2C_BUSBUSY ) == 0)
-//  {
-
-    i2cSetSlaveAdd(i2cREG2, addr);
-    i2cSetDirection(i2cREG2, I2C_TRANSMITTER);
-    i2cSetCount(i2cREG2, 3);
-    i2cSetMode(i2cREG2, I2C_MASTER);
-    i2cSetStop(i2cREG2);
-    i2cSetStart(i2cREG2);
-    i2cSendByte(i2cREG2, reg);
-    i2cSend(i2cREG2, 2, data);
-    while(i2cIsBusBusy(i2cREG2) == true);
-    while(i2cIsStopDetected(i2cREG2) == 0);
-    i2cClearSCD(i2cREG2);
-
-      //while (I2C_getStatus(i2c) == i2cTransferInProgress)
-      //while (i2c_IsBusBusy(i2c) == true)
-      //{
-        /* Enter EM1 while waiting for I2C interrupt */
-        //if(MAX_RETRIES < retries++)break;
-
-        //EMU_EnterEM1();
-        /* Could do a timeout function here. */
-      //}
-//  }
-//  else
-//  {
-      //Busy is busy return a error
-//      error = -1;
-//  }
-  //return((int)I2C_getStatus(i2c));
-  return i2cIsBusBusy(i2cREG2);
+  return i2c_Send(i2c, addr, 2, &data);
```

Comment:
I assume this is defined in i2c_io.h? I looked and wasn't able to find that file, so can't really finish the review unless I see that code. Side note: i2c_Send and i2c_Receive sound very similar to i2cSend and i2cReceive and might be easily confused. Might suggest making the name a bit more verbose. These new functions use interrupts, right?

Reply:
Check out https://github.com/AlbertaSat/ex2_obc_software/pull/50 (ideally this code would be linked as a static library, but alas).

These functions should replace the existing ones, and I'd prefer not making the names more verbose (the high level function name shouldn't/needn't reveal implementation details). If there are still i2c send/receive functions we should probably cut them out.

Reply:
I agree the names can be confusing. What would you suggest for a better name?

Reply:
Perhaps something like i2c_Send_msg and i2c_Receive_msg or i2c_msg_Send and i2c_msg_Receive?

Thanks for pointing me to that other PR. Implementation looks good